__tests__/app.test.js

```diff
@@ -39,8 +42,49 @@ describe("/api", () => {
       .get("/api")
       .expect(200)
       .then(({ body }) => {
-        console.log(endpointsObject)
         expect(body).toEqual(endpointsObject);
       });
   });
 });
+describe("/api/articles/:article_id", () => {
+  test("GET: 200 and responds with an article object with the following properties: author, title, article_id, body, topic, created_at, votes, article_iage_url", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        const { article } = body;
+        expect(article).toEqual(
+          expect.objectContaining({
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            author: "butter_bridge",
+            body: "I find this existence challenging",
+            created_at: expect.any(String),
+            votes: 100,
+            article_img_url:
+              "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          })
+        );
+      });
+  });
+})
+describe("Errors on /api/articles/:article_id", () => {
+  test("GET: 404 article should respond with Not found", () => {
+    return request(app)
+    .get("/api/random")
+    .expect(404)
+    .then((res) => {
+      expect(res.body).toEqual({ msg: "Not Found" })
+    })
+
+  });
+  test("GET: 404 and responds with an appropriate status and error message when given a valid but non existent query ", () => {
+    return request(app)
+      .get("/api/articles/1000")
+      .then((res) => {
+        expect(res.status).toBe(404);
+        expect(res.body).toEqual({ msg: "Not Found" });
+      });
+  });
+});
```

Comment:
An id of 1000 should absolutely be a 404 not found, but it's worth reconsidering about what should be sent back if the user requested an article with the id of 'random'. Is that a valid request? 404s are for 'valid datatype but non existing', how is this different from a request for the id of random?

As for implementing this - don't forget that the psql query itself will send back an error if you try to make a request of the wrong datatype, so we should be able to make use of this!